### PR TITLE
Error Instances Table: Add textual representation column and filtering based on Venn diagram and bar char clicks

### PR DESCRIPTION
**Pull request by ilmarinen**
1. Add a textual representation column in the error instances table.
2. Filter the error instances based on the user selections within the Venn diagram and bar chart.

### CODE REVIEW

**Review by jlema — CHANGES_REQUESTED** [review]
The histogram seems to be working correctly - as I click on each class the table is filtered (ground truth) by that class.

The venn diagram does not seem to be responding as expected. When I click on any of the areas the top few instances are the same. Also the percentages seem to be wrong as they add up to more than 100%.

1. Progress:
image
2. Regress:
image
3. Common:
image

Percentages adding up to more than 100%: 94% + 53% + 59%
image
image
image

**Review by jlema — APPROVED** [review]
Approved, note #55, #56, #58

File: backwardcompatibilityml/helpers/training.py [hunk 1]
```diff
@@ -313,7 +313,8 @@ def get_error_instance_indices(model, batched_evaluation_data, batched_evaluatio
         return model_diff.nonzero().view(-1).tolist()
 
 
-def get_all_error_instance_indices(h1, h2, batched_evaluation_data, batched_evaluation_target,
+def get_all_error_instance_indices(h1, h2, batch_ids, batched_evaluation_data, batched_evaluation_target,
+                                   get_instance_metadata=None,
```
Comment (Nking92) [inline backwardcompatibilityml/helpers/training.py:317]: add get_instance_metadata to docstring

File: backwardcompatibilityml/helpers/training.py [hunk 2]
```diff
@@ -487,6 +496,7 @@ def compatibility_scores(h1, h2, dataset, device="cpu"):
 
 
 def evaluate_model_performance_and_compatibility_on_dataset(h1, h2, dataset, performance_metric,
+                                                            get_instance_metadata=None,
```
Comment (Nking92) [inline backwardcompatibilityml/helpers/training.py:499]: add get_instance_metadata to docstring

File: backwardcompatibilityml/sweep_management.py [hunk 3]
```diff
@@ -66,6 +66,7 @@ def __init__(self, folder_name, number_of_epochs, h1, h2, training_set, test_set
                  performance_metric=model_accuracy,
                  get_instance_data_by_id=None,
                  get_instance_label_by_id=None,
+                 get_instance_metadata=None,
```
Comment (Nking92) [inline backwardcompatibilityml/sweep_management.py:69]: Add get_instance_metadata to the class docstring

File: widget/ErrorInstancesTable.tsx [hunk 4]
```diff
@@ -54,27 +55,43 @@ class ErrorInstancesTable extends Component<ErrorInstancesTableProps, ErrorInsta
...
+        maxWidth: 100,
+        isResizable: false
+      },
       { key: 'h1Prediction', name: 'h1 Prediction', fieldName: 'h1_prediction', minWidth: 100, maxWidth: 100, isResizable: false },
       { key: 'h2Prediction', name: 'h2 Prediction', fieldName: 'h2_prediction', minWidth: 100, maxWidth: 100, isResizable: false },
       { key: 'groundTruth', name: 'Ground Truth', fieldName: 'ground_truth', minWidth: 100, maxWidth: 100, isResizable: false },
     ];
 
+    var errorInstances = this.props.selectedDataPoint.error_instances;
+    console.log(this.props.filterInstances);
```
Comment (Nking92) [inline widget/ErrorInstancesTable.tsx:81]: FYI - you checked in console.log statements ( not sure if it was intentional)

**Review by Nking92 — COMMENTED** [review]
Looks good, just noticed a few places where docs need updates

**Review by Nking92 — APPROVED** [review]

File: widget/ErrorInstancesTable.tsx (hunk 4 above)
Comment (ilmarinen) [reply widget/ErrorInstancesTable.tsx:81]: Thanks.

**Review by nushib — APPROVED** [review]

**Pull request merged**